force-app/main/default/lwc/lookUp/lookUp.js

```diff
@@ -0,0 +1,98 @@
+import { LightningElement, wire, api, track } from 'lwc';
+import getRecordList from '@salesforce/apex/LookUpController.getRecordList';
+
+export default class Lookups extends LightningElement {
+
+    @api
+    sObjectApiName;
+
+    @api
+    addInfoFieldApiName = '';
+
+    @track
+    searchKey = '';
+
+    @track
+    selectedRecord;
+
+    @track
+    records;
+
+    @track 
+    error;
+
+    @track 
+    flag = false;
+
+    @wire(getRecordList, {
+        sObjectName : '$sObjectApiName', 
+        addInfo : '$addInfoFieldApiName',
+        searchKey : '$searchKey'
+    })
+    getRecords({
+        data, error
+    }) {
+        if(data) {
+            if(data.length > 0) {
+                this.records = data.map(element => {
+                    const {Id, Name } = element;
+                    return {Id, Name, AddInfoData : element[this.addInfoFieldApiName]};
+                });
+            } else {
+                this.records = null;
```

Comment:
@TsSlava remove redundant logic, you can do length check in the place where you are using records